docstore/memdocstore/mem.go

```diff
@@ -399,16 +405,34 @@ func (c *collection) checkRevision(arg driver.Document, current storedDoc) error
 
 // getAtFieldPath gets the value of m at fp. It returns an error if fp is invalid
 // (see getParentMap).
-func getAtFieldPath(m map[string]interface{}, fp []string) (interface{}, error) {
-	m2, err := getParentMap(m, fp, false)
-	if err != nil {
-		return nil, err
+func getAtFieldPath(m map[string]interface{}, fp []string, nested bool) (result interface{}, err error) {
+
+	var get func(m interface{}, name string) interface{}
+	get = func(m interface{}, name string) interface{} {
+		switch concrete := m.(type) {
+		case map[string]interface{}:
+			return concrete[name]
+		case []interface{}:
+			if !nested {
+				return nil
+			}
+			result := []interface{}{}
+			for _, e := range concrete {
+				result = append(result, get(e, name))
+			}
+			return result
```

Comment:
So if there are no matches, this will return an empty slice, which is not caught by the "== nil" check below so it will be returned.

Is that what you want?

Reply:
Done